### PR TITLE
Roa Rpki Validator From Internal

### PR DESCRIPTION
**Pull request by MiguelAHM**

### CODE REVIEW

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package net.ripe.db.whois.query.rpki;
```
Comment (eshryane) [inline whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java:1]: Don't put these classes in `whois-query` that package is for querying *whois*, instead put them in `whois-commons`

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@Profile({WhoisProfile.DEPLOYED})
+public class RoutinatorDataProvider implements RpkiDataProvider{
+
+    private static final int CLIENT_CONNECT_TIMEOUT = 20_000;
+    private static final int CLIENT_READ_TIMEOUT = 120_000;
+
+    private final Client client;
+    private final String rpkiBaseUrl;
+
+    public RoutinatorDataProvider(@Value("${rpki.base.url}") final String rpkiBaseUrl) {
```
Comment (eshryane) [inline whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java:26]: This client becomes *mandatory* if you have to set `rpki.base.url`. Instead make it optional and fail if the client is used.

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/WhoisRoaChecker.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+import net.ripe.db.whois.common.rpsl.RpslObject;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static net.ripe.db.whois.query.rpki.ValidationStatus.INVALID;
+
+public class WhoisRoaChecker extends RpkiRoaChecker {
```
Comment (eshryane) [inline whois-query/src/main/java/net/ripe/db/whois/query/rpki/WhoisRoaChecker.java:11]: Is this class a Component?

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RpkiService.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import net.ripe.db.whois.common.ip.Ipv6Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class RpkiService {
```
Comment (eshryane) [inline whois-query/src/main/java/net/ripe/db/whois/query/rpki/RpkiService.java:17]: Is this class a Component?

**Comment by eshryane** [comment]
One more thing, maybe reduce `roas.json` in size, it's 18MB

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RpkiService.java (hunk 4 above)
Comment (MiguelAHM) [reply whois-query/src/main/java/net/ripe/db/whois/query/rpki/RpkiService.java:17]: I would say Service

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java (hunk 1 above)
Comment (MiguelAHM) [reply whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java]: moved to whois-commons

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java]: I made it optional and return a null with an error log.
Upper class is throwing an error if it gets a null from that call.
I marked the method as nullable as well

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/WhoisRoaChecker.java (hunk 3 above)
Comment (MiguelAHM) [reply whois-query/src/main/java/net/ripe/db/whois/query/rpki/WhoisRoaChecker.java]: added a component

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java (hunk 2 above)
Comment (eshryane) [reply whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java]: Thanks

File: whois-query/src/main/java/net/ripe/db/whois/query/rpki/RpkiService.java (hunk 4 above)
Comment (eshryane) [reply whois-query/src/main/java/net/ripe/db/whois/query/rpki/RpkiService.java]: Looks good thanks

**Review by eshryane — APPROVED** [review]

**Pull request merged**